### PR TITLE
feat: add basic folder structure

### PR DESCRIPTION
**Pull request by anujakiran**
PR #1
Initial setup and basic code structure

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
+  "env": {
+    "commonjs": true,
+    "es2021": true,
```
Comment (Kiransukumaran) [inline .eslintrc.json:4]: Add remaining config from wl-conversation

File: .gitignore [hunk 2]
```diff
@@ -1,130 +1,3 @@
...
-.node_repl_history
-
-# Output of 'npm pack'
-*.tgz
-
-# Yarn Integrity file
-.yarn-integrity
-
-# dotenv environment variable files
+build
```
Comment (Kiransukumaran) [inline .gitignore:1]: Why this changes were removed ?

File: package.json [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    "dev": "nodemon src/index.ts",
+    "start": "node --env-file=.env build/index.js",
+    "build": "tsc",
+    "lint": "eslint src/**/*.ts --fix",
+    "prepare": "husky"
+  },
+  "author": "",
+  "license": "ISC",
+  "dependencies": {
+    "dotenv": "^16.4.5",
```
Comment (Kiransukumaran) [inline package.json:20]: Remove dotenv as it is not needed in 20.xx

File: package.json [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    "express": "^4.19.2",
+    "ts-node": "^10.9.2"
```
Comment (Kiransukumaran) [inline package.json]: dev dependency

File: src/app.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import express, { Request, Response } from 'express';
+import router from './router';
+const app = express();
```
Comment (Kiransukumaran) [inline src/app.ts:7]: add new line after line 2

File: src/app.ts [hunk 6]
```diff
@@ -0,0 +1,10 @@
...
+
+app.get('/', (_req: Request, res: Response) => {
```
Comment (Kiransukumaran) [inline src/app.ts]: Remove this endpoint

File: src/app.ts [hunk 7]
```diff
@@ -0,0 +1,10 @@
...
+  res.send('Hiii!');
+});
+app.use(router);
```
Comment (Kiransukumaran) [inline src/app.ts:11]: Add required middlewares for security and cors

File: src/controller/statusController.ts [hunk 8]
```diff
@@ -0,0 +1,10 @@
+import { Request, Response } from 'express';
+import { version } from '../../package.json';
+import { responseHelperSuccess } from '../helper/responseHelper';
+
+const getStatus = (_req: Request, res: Response) => {
+  const response = { Environment: process.env.NODE_ENV, version: version };
```
Comment (Kiransukumaran) [inline src/controller/statusController.ts]: Add Application name
Environment => environment

File: src/controller/statusController.ts [hunk 9]
```diff
@@ -0,0 +1,10 @@
+import { Request, Response } from 'express';
+import { version } from '../../package.json';
+import { responseHelperSuccess } from '../helper/responseHelper';
+
+const getStatus = (_req: Request, res: Response) => {
```
Comment (Kiransukumaran) [inline src/controller/statusController.ts]: export const getStatus

File: src/helper/responseHelper.ts [hunk 10]
```diff
@@ -0,0 +1,10 @@
+import { Response } from 'express';
+
+const responseHelperSuccess = (res: Response, result: object) => {
+  res.status(200).json({
```
Comment (Kiransukumaran) [inline src/helper/responseHelper.ts]: Allow other success status codes as well like 201 created

File: src/helper/responseHelper.ts [hunk 11]
```diff
@@ -0,0 +1,10 @@
+import { Response } from 'express';
+
+const responseHelperSuccess = (res: Response, result: object) => {
```
Comment (Kiransukumaran) [inline src/helper/responseHelper.ts]: export const

File: tsconfig.json [hunk 12]
```diff
@@ -0,0 +1,15 @@
+{
+  "compilerOptions": {
+    "target": "es2016",
```
Comment (Kiransukumaran) [inline tsconfig.json]: refer to wl-conversation

File: tsconfig.json [hunk 13]
```diff
@@ -0,0 +1,15 @@
...
+    "module": "commonjs",
+    "rootDir": "./src",
+    "outDir": "build",
+    "esModuleInterop": true,
+    "resolveJsonModule": true,
+    "forceConsistentCasingInFileNames": true,
+    "strict": true,
+    "skipLibCheck": true
+  },
+  "include": ["src/**/*.ts", "src/*.ts"],
```
Comment (Kiransukumaran) [inline tsconfig.json]: ?????

File: src/helper/responseHelper.ts (hunk 11 above)
Comment (Kiransukumaran) [inline src/helper/responseHelper.ts]: use more sensible type

File: src/helper/responseHelper.ts [hunk 14]
```diff
@@ -0,0 +1,10 @@
...
+  res.status(200).json({
+    status: 'OK',
```
Comment (Kiransukumaran) [inline src/helper/responseHelper.ts]: lowercase

**Review by Kiransukumaran — CHANGES_REQUESTED** [review]
Please add the following as well
1) Update the readme file with local setup information and details

File: .gitignore (hunk 2 above)
Comment (anujakiran) [reply .gitignore:1]: I thought it was unintentional. Please confirm if it needs to be added back

File: src/helper/responseHelper.ts (hunk 14 above)
Comment (anujakiran) [reply src/helper/responseHelper.ts]: done

File: src/helper/responseHelper.ts (hunk 11 above)
Comment (anujakiran) [reply src/helper/responseHelper.ts]: done

File: tsconfig.json (hunk 13 above)
Comment (anujakiran) [reply tsconfig.json]: done

File: tsconfig.json (hunk 12 above)
Comment (anujakiran) [reply tsconfig.json]: ok

File: src/helper/responseHelper.ts (hunk 11 above)
Comment (anujakiran) [reply src/helper/responseHelper.ts]: changed

File: package.json (hunk 3 above)
Comment (anujakiran) [reply package.json:20]: not possible

File: .gitignore (hunk 2 above)
Comment (Kiransukumaran) [reply .gitignore:1]: for now lets keep it like this

**Review by Kiransukumaran — APPROVED** [review]

**Pull request merged**